packages/react-components/src/components/MessageThread.tsx

```diff
@@ -601,6 +601,12 @@ export type MessageThreadProps = {
    * this function will be called with the data inside `fileSharingMetadata` key.
    */
   fileDownloadHandler?: FileDownloadHandler;
+
+  /**
+   * Optional function to provide customized date format.
+   *
+   */
+  customizedFormatTimestampForChatMessage?: (messageDate: Date) => string;
```

Comment:
can just call this prop "dateTimeFormat"

Reply:
this is really a callback that gives a dateTime and wants to return a string

I am thinking something like on__________ would be a better naming convention? 
onRenderDateTime doesn't seem quite right since it kind of implies returning a JSX.Element based on our other onRenders?

I think it might be good to have the same named prop in ComponentLocale and everywhere else in the library?
@PorterNan  thoughts?